### PR TITLE
Making Ly animations easily extendable

### PR DESCRIPTION
**Pull request by SladeGetz**
I consolidated the animation functions into their own .c and .h file (removed from draw) and made it much easier to add new animations by simply modifying `#define ANIMATION_NUM ...` on line 32 of `src/animations.c` and adding your animation init, run, and free functions to the arrays `static const void* ANIM_INITS[]`, `static const void* ANIM_RUNS[]`, and `static const void* ANIM_FREES[]` on lines 43, 47, and 51, respectively. Additionally, the animation selection is now O(1), instead of O(n).

### CODE REVIEW

File: src/animations.c [hunk 1]
```diff
@@ -0,0 +1,419 @@
...
+        int i;
+        if ((i = config.animation) < ANIMATION_NUM)
+        {
+            void (*fun_ptr)(struct term_buf*);
+            fun_ptr = ANIM_INITS[i];
+            (*fun_ptr)(buf);
+        }
+	}
+}
+
```
Comment (AnErrupTion) [inline src/animations.c:417]: I'm perosnally not a huge fan of this. In my opinion, it decreases code readability too much for what is a minor improvement.

File: src/draw.h [hunk 2]
```diff
@@ -59,9 +34,12 @@ struct term_buf
...
 
-	union anim_state astate;
+    // void pointer allows not
+    // explicitly declaring animations
+    void* astate;
 };
 
+
 void draw_init(struct term_buf* buf);
 void draw_free(struct term_buf* buf);
```
Comment (AnErrupTion) [inline src/draw.h:44]: ``draw_free()`` doesn't seem to be used anymore, perhaps it could be removed?

**Review by AnErrupTion — CHANGES_REQUESTED** [review]
Hey! I personally like your chances, although I had 2 things to say. Could you rebase your PR on master too please? Thanks!

**Comment by Flixnore** [comment]
Any plans to continue this? Just found Ly and hoping to make my own animations.

**Comment by SladeGetz** [comment]
Moved to #541

**Pull request closed without merging**